### ISSUE TITLE
Instantly Crashs on "Initialization" for "InterstitialAd" & "RewardVideoBasedAd"

### ISSUE THREAD

**Issue by tashfiq103** [googleads/googleads-mobile-unity#553]
Thank you so much for taking your valuable time to take a look into the following issue.

I have attached the "unityPackage" by which you can test it on your editor & device. Also, the following script shows how it was ported to physical device for testing.

UnityPakage : https://drive.google.com/open?id=0B-YWb5BzJ1ZeSkFyN2VKTGtqY0E
"Video" showing crash report : https://drive.google.com/open?id=0B-YWb5BzJ1ZeVWRfYlpjdG4zd2c
DeviceID : https://drive.google.com/open?id=0B-YWb5BzJ1Zed29yT1NJcnl4cEU

Additional Information:

    Unity Version ; 2017.1.0f3
    Admob SDK  : Google Mobile Ads Unity Plugin - v3.8.0
    Android Device : Asus Zenfone - 5
    OS Version : 5.0 (Lolipop)

```C#
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using GoogleMobileAds;
using GoogleMobileAds.Api;

public class TestScript : MonoBehaviour {

	InterstitialAd interstitial;

	private string debugMessege;
	// Use this for initialization
	void Start () {

	}
	
	private void mInitializeInterstetialAd(){

		debugMessege = "Before Initialization"; 

		#if UNITY_ANDROID
		// InterstitialAd testID from admob
		string adUnitId = "ca-app-pub-3940256099942544/1033173712";
		#elif UNITY_IPHONE
		string adUnitId = "INSERT_IOS_INTERSTITIAL_AD_UNIT_ID_HERE";
		#else
		string adUnitId = "unexpected_platform";
		#endif


		// Initialize an InterstitialAd.
		interstitial = new InterstitialAd(adUnitId);
		debugMessege += "\nInterstitialAd Initialized";

		// Create an empty ad request.

		AdRequest request = new AdRequest.Builder()
		.AddTestDevice(AdRequest.TestDeviceSimulator)
		.AddTestDevice("A1803A2F089A97C5")
		.AddTestDevice("357996068713245")
		.Build();

		debugMessege += "\nAfter AdRequest";

		// Load the interstitial with the request.
		interstitial.LoadAd(request);
		debugMessege += "\nLoad InterstetialAd";
	}

	void OnGUI(){

		float mHeight = Screen.height / 10.0f;

		if (GUI.Button (new Rect(0.0f, 0.0f, Screen.width, mHeight),"Initialize")) {

			mInitializeInterstetialAd ();	
		}

		if (GUI.Button (new Rect(0.0f, mHeight, Screen.width, mHeight),"ShowAd")) {
			
			if(interstitial.IsLoaded())
				interstitial.Show ();
		}

		GUI.Label (new Rect (0.0f, mHeight * 2, Screen.width, mHeight * 2), debugMessege);
	}
}
```

**Comment by rampara** [comment]
@tashfiq103 can you provide a logcat log of the crash?

**Comment by QuentinPix** [comment]
Hi,

Jumping into this thread as I have exactly the same issue, with a similar c# code.

Here is the LogCat : 

```
12-08 16:00:31.461 19405-19405/? D/DynamitePackage: Instantiating com.google.android.gms.ads.ChimeraAdOverlayCreatorImpl
12-08 16:00:31.467 19405-19405/? I/Ads: Ad opening.
12-08 16:00:31.470 3289-10134/? D/ActivityManager: setRequestedOrientation() to 7 for ActivityRecord{1244daf u0 com.Hidden.Ink/com.google.android.gms.ads.AdActivity t251} from pid=19405, uid=10222
12-08 16:00:31.470 3289-10134/? E/ActivityManager: Activity Manager Crash. UID:10222 PID:19405 TRANS:63
                                                   java.lang.IllegalStateException: Only fullscreen activities can request orientation
                                                       at com.android.server.am.ActivityRecord.setRequestedOrientation(ActivityRecord.java:2263)
                                                       at com.android.server.am.ActivityManagerService.setRequestedOrientation(ActivityManagerService.java:5919)
                                                       at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:1102)
                                                       at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3477)
                                                       at android.os.Binder.execTransact(Binder.java:692)
12-08 16:00:31.471 19405-19405/? D/AndroidRuntime: Shutting down VM
12-08 16:00:31.471 19405-19405/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.Hidden.Ink, PID: 19405
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{com.Hidden.Ink/com.google.android.gms.ads.AdActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046)
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688)
                                                       at android.os.Handler.dispatchMessage(Handler.java:105)
                                                       at android.os.Looper.loop(Looper.java:164)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                    Caused by: java.lang.IllegalStateException: Only fullscreen activities can request orientation
                                                       at android.os.Parcel.readException(Parcel.java:1950)
                                                       at android.os.Parcel.readException(Parcel.java:1888)
                                                       at android.app.IActivityManager$Stub$Proxy.setRequestedOrientation(IActivityManager.java:5918)
                                                       at android.app.Activity.setRequestedOrientation(Activity.java:5762)
                                                       at com.google.android.gms.ads.internal.overlay.d.a(:com.google.android.gms.DynamiteModulesA@11951440:182)
                                                       at com.google.android.gms.ads.internal.overlay.d.b(:com.google.android.gms.DynamiteModulesA@11951440:216)
                                                       at com.google.android.gms.ads.internal.overlay.d.a(:com.google.android.gms.DynamiteModulesA@11951440:76)
                                                       at com.google.android.gms.ads.internal.overlay.client.c.onTransact(:com.google.android.gms.DynamiteModulesA@11951440:8)
                                                       at android.os.Binder.transact(Binder.java:622)
                                                       at com.google.android.gms.internal.zzed.zzb(Unknown Source:7)
                                                       at com.google.android.gms.internal.zzwl.onCreate(Unknown Source:8)
                                                       at com.google.android.gms.ads.AdActivity.onCreate(Unknown Source:28)
                                                       at android.app.Activity.performCreate(Activity.java:6998)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1230)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2899)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046) 
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                       at android.os.Looper.loop(Looper.java:164) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
12-08 16:00:31.472 19405-19405/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.Hidden.Ink, PID: 19405
                                                   java.lang.Error: FATAL EXCEPTION [main]
                                                   Unity version     : 5.6.0f3
                                                   Device model      : OnePlus ONEPLUS A3003
                                                   Device fingerprint: OnePlus/OnePlus3/OnePlus3:8.0.0/OPR6.170623.013/10250816:user/release-keys
                                                   
                                                   Caused by: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.Hidden.Ink/com.google.android.gms.ads.AdActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046)
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688)
                                                       at android.os.Handler.dispatchMessage(Handler.java:105)
                                                       at android.os.Looper.loop(Looper.java:164)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                    Caused by: java.lang.IllegalStateException: Only fullscreen activities can request orientation
                                                       at android.os.Parcel.readException(Parcel.java:1950)
                                                       at android.os.Parcel.readException(Parcel.java:1888)
                                                       at android.app.IActivityManager$Stub$Proxy.setRequestedOrientation(IActivityManager.java:5918)
                                                       at android.app.Activity.setRequestedOrientation(Activity.java:5762)
                                                       at com.google.android.gms.ads.internal.overlay.d.a(:com.google.android.gms.DynamiteModulesA@11951440:182)
                                                       at com.google.android.gms.ads.internal.overlay.d.b(:com.google.android.gms.DynamiteModulesA@11951440:216)
                                                       at com.google.android.gms.ads.internal.overlay.d.a(:com.google.android.gms.DynamiteModulesA@11951440:76)
                                                       at com.google.android.gms.ads.internal.overlay.client.c.onTransact(:com.google.android.gms.DynamiteModulesA@11951440:8)
                                                       at android.os.Binder.transact(Binder.java:622)
                                                       at com.google.android.gms.internal.zzed.zzb(Unknown Source:7)
                                                       at com.google.android.gms.internal.zzwl.onCreate(Unknown Source:8)
                                                       at com.google.android.gms.ads.AdActivity.onCreate(Unknown Source:28)
                                                       at android.app.Activity.performCreate(Activity.java:6998)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1230)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2899)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3046) 
                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                       at android.os.Looper.loop(Looper.java:164) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6809) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
12-08 16:00:31.472 19405-19405/? D/AppTracker: App Event: crash

```
I'm quite new to coding, but this is the only crash I can't seem to resolve by myself so far...

Cheers,

Quentin

**Comment by rampara** [comment]
@QuentinPix the error you're running into is cased by a bug in the Google Mobile Ads SDK when building for API 27. A fix for this bug has been implemented will be included in an upcoming release of the SDK. For now, as a workaround, you can target < API 27.

**Comment by QuentinPix** [comment]
@rampara Thanks for the info. Is there a clean way to downgrade ? I have read about the build.gradle, is this the best way to do so in unity ? Cheers, Quentin

**Comment by rampara** [comment]
@QuentinPix you can set the `Target API Level` to the appropriate value under `Other Settings` for Android Player Settings.

**Comment by Snakes128** [comment]
Hi rampara,

I have the same problem with the last package "Google Mobile Ads Unity Plugin v3.10.0" and 8.0 Android SDK with the last API "27"

**Comment by rampara** [comment]
@Snakes128 this crash isn't caused by this plugin but rather the GMA Android SDK. Can you try updating the `GoogleMobileAdsDependencies` file at the `Assets/GoogleMobileAds/Editor` directory of your project use the latest version of the GMA Android SDK (11.6.2), which should include a fix for this issue.

**Comment by dariuszpietrala** [comment]
I have a lot of info about crashes (unfortunatelly I don't have android 8.0, so I don't have logcat logs and I can't seem to find any crash reports from those users). All happen on Android 8.0 when trying to show interstitial. Plugin v3.10.0, GMA Android SDK 11.6.2, Unity 5.6.5f1, target API "highest installed" and it's API 27. Any ideas?

**Comment by rampara** [comment]
@dariuszpietrala I was mistaken in that a fix for this issue was included in GMA Android SDK 11.6.2. The fix will actually be included in the next release of the GMA Android SDK. As a workaround, you can target an API < 27.

**Comment by dariuszpietrala** [comment]
Thank you. Seems to work fine. Built with API 26 and I already got info from one of the users that it works.

**Comment by rampara** [comment]
@dariuszpietrala glad your issue has been resolved. With the next release of the GMA Android SDK, this workaround won't be required.

**Comment by planethcom** [comment]
@rampara In which update will the fix be included? In 11.8.0 it's still not fixed

**Comment by rampara** [comment]
@planethcom the fix will be in the upcoming release following 11.8.0.

**Comment by planethcom** [comment]
@rampara Thanks for your answer. Just to avoid confusion, when you say 11.8.0, what exact SDK do you mean? Google Mobile Ads SDK 11.8.0 (served through Google Play Services)? If so ... We're already using 11.8.0 since December, but the problem is still there when targeting API 27 (affected are only Android 8.0 devices, Android 8.1 works without problems)

**Comment by rampara** [comment]
@planethcom i'm sorry for the confusion. The fix for this issue is not included in Google Play services 11.8.0. It will be included in the upcoming release of Google Play services.

**Comment by rampara** [comment]
Closing as this isn't an issue with the plugin but the Android GMA SDK.
